config/routes.rb

```diff
@@ -0,0 +1,16 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  root 'tasks#index'
+
+  get '/tasks', to: 'tasks#index', as: "tasks"
+  get '/tasks/new', to: 'tasks#new', as: "new_task"
+  post '/tasks', to: 'tasks#create'
+
+  get '/tasks/:id', to: 'tasks#show', as: "task"
+  get '/tasks/:id/edit', to: 'tasks#edit', as: "edit_task"
+  patch '/tasks/:id', to: 'tasks#update'
+  put '/tasks/:id', to: 'tasks#mark_complete'
+
```

Comment:
Since the `mark_complete` route is non-RESTful anyway, it might be more clear to give it a more descriptive path, something like `put '/tasks/:id/mark_complete'`.

As is you're relying on the difference between PATCH and PUT, which doesn't give a lot of info for example when looking at the server output.